### PR TITLE
adding details regarding using firewalld instead of the iptables

### PR DESCRIPTION
**Pull request by rlopez133**
Adding some documentation on how to use firewalld instead of the iptables command provided in the original document.

### CODE REVIEW

File: Documentation/dev/libvirt-howto.md [hunk 1]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
   -j ACCEPT -m comment --comment "Allow insecure libvirt clients"
 ```
 
+If using `firewalld`, simply optain the name of the existing active zone which
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: Can you strip the trailing whitespace here (and in other places) to make Git happy:

```console
$ git log -2 --check --oneline origin/pr/284
c66b9c9 fixed a small typo
2ce46b2 adding details regarding using firewalld instead of the iptables
Documentation/dev/libvirt-howto.md:88: trailing whitespace.
+If using `firewalld`, simply optain the name of the existing active zone which 
Documentation/dev/libvirt-howto.md:99: trailing whitespace.
+unless otherwise specified. 
```

File: Documentation/dev/libvirt-howto.md [hunk 2]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
   -j ACCEPT -m comment --comment "Allow insecure libvirt clients"
 ```
 
+If using `firewalld`, simply optain the name of the existing active zone which 
+can be used to integrate the appropriate source and ports to allow connections from
+the IP range used by your cluster nodes. An example is shown below.
+
+````
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: The example looks like a shell session, in which case, can you add a prompt and:

    ```console

for highlighting?  That would be:

```console
$ sudo firewall-cmd --get-active-zones
FedoraWorkstation
  interfaces: enp0s25 tun0
```

File: Documentation/dev/libvirt-howto.md [hunk 3]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
...
+sudo firewall-cmd --get-active-zones
+FedoraWorkstation
+  interfaces: enp0s25 tun0
+````
+With the name of the active zone, include the source and port to allow connections
+from the IP range used by your cluster nodes. The default subnet is `192.168.124.0/24`
+unless otherwise specified. 
+
+````
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: You don't have any output here, and for shell examples without output we prefer no prompts and:

    ```sh

syntax highlighting (more on this in #27).

File: Documentation/dev/libvirt-howto.md [hunk 4]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
...
+Verification of the source and port can be done listing the zone
+
+````
+sudo firewall-cmd --zone=FedoraWorkstation --list-ports
+sudo firewall-cmd --zone=FedoraWorkstation --list-sources
+````
+
+NOTE: When the firewall rules are no longer needed, a simple `reload` of `firewalld`
+will remove the changes made as they were not permanently added. For persistence,
+include the --permanent to the commands that add-source and add-port.
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: Can you backtick the literal `--permanent`?

File: Documentation/dev/libvirt-howto.md [hunk 5]
```diff
@@ -150,6 +179,13 @@ tectonic destroy --dir=$CLUSTER_NAME
 ```
 Be sure to destroy, or else you will need to manually use virsh to clean up the leaked resources. The [`virsh-cleanup`](../../scripts/maintenance/virsh-cleanup) script may help with this, but note it will currently destroy *all* libvirt resources.
 
+Ensure to restart your `firewalld` to remove the source and opened 16509/tcp port
+if no longer necessary using the following command.
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: Maybe reword to something like:

> Restart `firewalld` to remove your temporary changes:

or:

> With the cluster removed, you no longer need to allow libvirt nodes to reach your `libvirtd`.  Restart `firewalld` to remove your temporary changes:

File: Documentation/dev/libvirt-howto.md [hunk 6]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
...
+````
+
+Verification of the source and port can be done listing the zone
+
+````
+sudo firewall-cmd --zone=FedoraWorkstation --list-ports
+sudo firewall-cmd --zone=FedoraWorkstation --list-sources
+````
+
+NOTE: When the firewall rules are no longer needed, a simple `reload` of `firewalld`
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: Maybe "a simple `reload` of `firewalld`" -> `firewalld --reload`.

**Comment by wking** [comment]
Ok, I think I'm done :p.  This looks great, thanks :).

**Comment by rlopez133** [comment]
let me know if this all looks good then i can squash

File: Documentation/dev/libvirt-howto.md [hunk 7]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
   -j ACCEPT -m comment --comment "Allow insecure libvirt clients"
 ```
 
+If using `firewalld`, simply optain the name of the existing active zone which
+can be used to integrate the appropriate source and ports to allow connections from
+the IP range used by your cluster nodes. An example is shown below.
+
+```console
+sudo firewall-cmd --get-active-zones
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: With `console` highlighting (which I think we want because you're showing output), you need [a `$` prompt here][1].

[1]: https://github.com/openshift/installer/pull/284#discussion_r218938492

Comment (rlopez133) [reply Documentation/dev/libvirt-howto.md]: do you not want prompts on the others as well?

Comment (wking) [reply Documentation/dev/libvirt-howto.md]: > do you not want prompts on the others as well?

Prompts belong in `console` blocks (to differentiate between commands and output), but not in `sh` blocks (because they're not part of the shell input).  In this project, `sh` was more popular where we don't need to show output, so that's the pattern I'm trying to stick to (more on this in #27).

File: Documentation/dev/libvirt-howto.md [hunk 8]
```diff
@@ -85,6 +85,35 @@ iptables -I INPUT -p tcp -s 192.168.124.0/24 -d 192.168.124.1 --dport 16509 \
...
+```
+
+Verification of the source and port can be done listing the zone
+
+```sh
+sudo firewall-cmd --zone=FedoraWorkstation --list-ports
+sudo firewall-cmd --zone=FedoraWorkstation --list-sources
+```
+
+NOTE: When the firewall rules are no longer needed, `firewalld --reload`
```
Comment (wking) [inline Documentation/dev/libvirt-howto.md]: Oops, looks like this should have been `firewall-cmd --reload`, based on the example below :/.
